crates/flux-refineck/src/type_env.rs

```diff
@@ -886,20 +888,30 @@ struct TypeEnvBind {
     ty: String,
 }
 
+fn loc_string(local_names: &UnordMap<Local, Symbol>, loc: &Loc) -> Option<String> {
+    if let Loc::Local(local) = loc {
+        let name = local_names.get(local)?;
+        return Some(format!("{}", name));
+    }
+    Some(format!("{:?}", loc))
+}
+
 impl TypeEnvTrace {
-    pub fn new(genv: GlobalEnv, env: &TypeEnv) -> Self {
+    pub fn new(genv: GlobalEnv, local_names: &UnordMap<Local, Symbol>, env: &TypeEnv) -> Self {
         let mut bindings = vec![];
-        let cx = PrettyCx::default_with_genv(genv);
+        let cx = PrettyCx::default_with_genv(genv).hide_regions(true);
         env.bindings
             .iter()
             .filter(|(_, binding)| !binding.ty.is_uninit())
             .sorted_by(|(loc1, _), (loc2, _)| loc1.cmp(loc2))
             .for_each(|(loc, binding)| {
-                let loc = format!("{loc:?}");
-                let kind = format!("{:?}", binding.kind);
-                let ty = WithCx::new(&cx, binding.ty.clone());
-                let ty = format!("{:?}", ty);
-                bindings.push(TypeEnvBind { loc, kind, ty });
+                // filtering out bindings that do not correspond to source-level names
+                if let Some(loc) = loc_string(local_names, loc) {
+                    let kind = format!("{:?}", binding.kind);
+                    let ty = WithCx::new(&cx, binding.ty.clone());
+                    let ty = format!("{:?}", ty);
+                    bindings.push(TypeEnvBind { loc, kind, ty });
+                }
```

Comment:
Maybe we can store both the `loc` and the (optional) var name and decide whether to show it to the client.